lib/Core/Executor.cpp

```diff
@@ -867,14 +867,62 @@ bool Executor::branchingPermitted(const ExecutionState &state) const {
   return true;
 }
 
+/// Helper method to get an Assignment for a state.  This method is called
+/// when we have reached the maximum number of forks allotted for a particular
+/// exploration.
+Assignment * generateAssignmentForDanglingState(std::map<ExecutionState*, Assignment*> &maxForksMap,
```

Comment:
Why is this needed? It's basically the CEX caching solver, could just that be used? Executor.cpp is long enough as it is. Also why `std::map` instead of `std::unordered_map`?

Reply:
I'd say this is more efficient, as you have one solution associated with each state.   In the CEX cache, you don't have this direct mapping from states to solutions (but we should have something like this!).

Reply:
Isn't `CexCachingSolver::cache` kind of a mapping between states and solutions? I guess the difference is `MapOfSets` vs `std::map` implementation and a pointer key vs set of expression key, but I'm not convinced optimizing these is worth this new implementation. 

For a direct mapping from states to solutions, wouldn't it be better to just store the `Assigment` directly into `ExecutionState`? I guess it uses a bit more memory when running under the memory limit, but running at the memory limit seems to be the more common case.

Reply:
Yes, storing an Assignment directly into an ExecutionState is what I'd do.

But getting back to this PR, I think it's good enough as it is.  The code is only exercised when MaxForks is reached, so it doesn't interfere with the core.  And since it doesn't interfere with the core, I don't think it's reasonable to ask for a completely different PR...

Reply:
But it would be a small change for this PR to use an Assignment field on ExecutionState and would lay some ground work for that PR, instead of introducing a third way.

Reply:
I guess I'm concerned how #1334 could fit into this, I opened it so that the discussion can be a bit more concrete. The PR can coexists quite easily, but they will have 2 different systems doing the same thing.